### PR TITLE
RECIPE APP

### PR DESCRIPTION
**Pull request by ndorrh**
This pull request consists of the following changes;
Developed the app according to the following requirements

# Project requirements 🎯 🎯
- ###  You should follow the layout of the wireframes provided. You should personalize the rest of the design including colors, typographies, spacings, etc.

- ### Login page and registration page:

- [ ] Should be built with Devise.

- ### Food list:

- [ ] Should display a list of food added by the logged-in user as in the [wireframe](https://github.com/microverseinc/curriculum-rails/raw/main/recipe-app/images/food_index.png) (for teams with only 2 people display also quantity of a given food).

- [ ] This should lead to a form that allows users to add new food.

- ### Recipes list:

- [ ] Should display a list of recipes created by the logged-in user as in the [wireframe](https://github.com/microverseinc/curriculum-rails/raw/main/recipe-app/images/recipe_index.png).

- [ ] This should lead to recipe details.

- [ ] If the user is the owner of the recipe, should allow the user to delete it.

- ### Public recipe list:

- [ ] Should display a list of all public recipes ordered by newest as in the [wireframe](https://github.com/microverseinc/curriculum-rails/raw/main/recipe-app/images/public_recipe_index.png).

- [ ] This should lead to recipe details.

- ### Recipe details:

- [ ] If the recipe is public or the user is the owner of the recipe, should display the recipe details as in the [wireframe](https://github.com/microverseinc/curriculum-rails/raw/main/recipe-app/images/recipe_food.png).

- [ ] If the user is the owner of the recipe, should lead to the form that allows the user to add new food.

- ### General shopping list view (only for a group that has less than 3 members):

- [ ] Should show the list of food that is missing for all recipes of the logged-in user as in the [wireframe](https://github.com/microverseinc/curriculum-rails/raw/main/recipe-app/images/general_shopping_list.png) (compare the list of food for all recipes with the general food list of that user).

- [ ] Should count the total food items and the total price of the missing food.

- ### Make sure there are no N+1 queries happening.

- ### Create a navigation menu that allows users to open all of the pages you created.

### CODE REVIEW

File: app/controllers/recipes_controller.rb [hunk 1]
```diff
@@ -0,0 +1,83 @@
+class RecipesController < ApplicationController
+  load_and_authorize_resource
+  before_action :set_recipe, only: %i[show edit update destroy]
+
+  # GET /recipes or /recipes.json
+  def index
+    @recipes = Recipe.all
```
Comment (Davidosky007) [inline app/controllers/recipes_controller.rb]: * As stated in the [project requirement](https://github.com/microverseinc/curriculum-rails/blob/main/recipe-app/buisness_requirements.md#:~:text=Make%20sure%20there%20are%20no%20N%2B1%20queries%20happening.) n + 1 issues should be handled , Kindly fix all n +1 issues 🙏

File: app/controllers/inventory_foods_controller.rb [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  def index
+    @inventory_foods = InventoryFood.all
+  end
+
+  # GET /inventory_foods/1 or /inventory_foods/1.json
+  def show; end
+
+  # GET /inventory_foods/new
+  def new
+    @inventory_food = InventoryFood.new
```
Comment (Davidosky007) [inline app/controllers/inventory_foods_controller.rb:15]: * Kindly fix this error i get when i try to create new inventory food 
image

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @ndorrh @iAmWaris97 @erickhansx 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
* All tests passed ✔️ 
* No linter errors 👍 
* Good looking readme file 👍 

## Required Changes ♻️
* Kindly fix all N +1 issues in your code 
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/inventory_foods_controller.rb (hunk 2 above)
Comment (ndorrh) [reply app/controllers/inventory_foods_controller.rb:15]: @Davidosky007 thanks for the review
I have fix the problem

File: app/controllers/recipes_controller.rb (hunk 1 above)
Comment (ndorrh) [reply app/controllers/recipes_controller.rb]: @Davidosky007 thanks for the review
I have fix the problem

**Review by danielufeli — APPROVED** [review]
Hi @Team,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

ChopApproveCopApproveGIF

## Highlights
- Nice commits messages✔️
- Clean Code ✔️

## Optional suggestions
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @danielufeli in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._ate_comments_counter`

**Pull request merged**